common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -337,28 +351,32 @@ private boolean processInvalidUrl(@NonNull final WebView view, @NonNull final St
             return true;
         }
 
-        if (url.toLowerCase(Locale.US).equals("about:blank")) {
+        if ("about:blank".equals(lowerCaseUrl)) {
             Logger.verbose(TAG, "It is an blank page request");
             return true;
         }
 
-        if (!url.toLowerCase(Locale.US).startsWith(AuthenticationConstants.Broker.REDIRECT_SSL_PREFIX)) {
-            String redactedUrl = "redacted";
-            try {
-                redactedUrl = StringExtensions.removeQueryParameterFromUrl(url);
-            } catch (final URISyntaxException e) {
-                // Best effort.
-            }
+        if (!lowerCaseUrl.startsWith(AuthenticationConstants.Broker.REDIRECT_SSL_PREFIX)) {
+            String redactedUrl = removeQueryParameters(url);
 
             Logger.error(TAG, "The webView was redirected to an unsafe URL: " + redactedUrl, null);
             returnError(ErrorStrings.WEBVIEW_REDIRECTURL_NOT_SSL_PROTECTED, "The webView was redirected to an unsafe URL.");
             view.stopLoading();
             return true;
         }
-
+        Logger.infoPII(TAG,"We are declining to override loading and redirect to invalid URL: '"
+                + removeQueryParameters(url) + "' the user's url pattern is '" + mRedirectUrl + "'");
```

Comment:
Should we log the actual url here? I mean the one with the query parameters still there

Reply:
I saw that we didn't to it above, and was just following that pattern.  Should we?

Reply:
I think it's better not too as we are not sure what data the qp's can have here, since this url is not known to us. Logging this invalid url should give us good info on what's wrong anyways.

Reply:
I think that we should be OK without it, so I'm leaving it out.